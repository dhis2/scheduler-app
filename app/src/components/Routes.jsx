import React from 'react';
import { Router, Route } from 'react-router-dom';
import history from '../utils/history';
import List from './jobOverview/List';
import EditJob from './jobContent/EditJob';
import AddJob from './jobContent/AddJob';

const style = {
    paddingTop: '100px',
    width: '100%',
    maxWidth: '1000px',
    margin: '0 auto',
};

const Routes = () => (
    <Router history={history}>
        <div style={style}>
            <Route exact path="/" component={List} />
            <Route path="/edit/:id" component={EditJob} />
            <Route path="/add" component={AddJob} />
        </div>
    </Router>
);

export default Routes;
