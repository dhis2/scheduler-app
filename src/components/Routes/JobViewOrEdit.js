import React from 'react'
import { NoticeBox } from '@dhis2/ui'
import { useParams } from 'react-router-dom'
import i18n from '@dhis2/d2-i18n'
import { useJobById } from '../../hooks/jobs'
import { Spinner } from '../Spinner'
import JobEdit from '../../pages/JobEdit'
import JobView from '../../pages/JobView'

const JobViewOrEdit = () => {
    const { id } = useParams()
    const { data, fetching, error } = useJobById(id)

    if (fetching) {
        return <Spinner />
    }

    if (error) {
        return (
            <NoticeBox error title={i18n.t('Could not load requested job')}>
                {i18n.t(
                    'Something went wrong whilst loading the requested job. Make sure it has not been deleted and try refreshing the page.'
                )}
            </NoticeBox>
        )
    }

    const { configurable } = data

    if (configurable) {
        return <JobEdit job={data} />
    } else {
        return <JobView job={data} />
    }
}

export default JobViewOrEdit
