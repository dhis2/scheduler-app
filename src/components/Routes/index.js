export { default as Routes } from './Routes.jsx'
