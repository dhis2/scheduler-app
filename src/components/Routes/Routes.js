import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'
import { Router } from 'react-router'
import JobList from '../../pages/JobList'
import JobAdd from '../../pages/JobAdd'
import QueueAdd from '../../pages/QueueAdd'
import QueueEdit from '../../pages/QueueEdit'
import history from '../../services/history'
import JobViewOrEdit from './JobViewOrEdit'

const Routes = () => (
    <Router history={history}>
        <Switch>
            <Redirect from="/edit/:id" to="/job/:id" />
            <Redirect from="/view/:id" to="/job/:id" />
            <Redirect from="/add" to="/job/add" />
            <Route exact path="/" component={JobList} />
            <Route path="/job/add" component={JobAdd} />
            <Route path="/job/:id" component={JobViewOrEdit} />
            <Route path="/queue/add" component={QueueAdd} />
            <Route path="/queue/:name" component={QueueEdit} />
        </Switch>
    </Router>
)

export default Routes
