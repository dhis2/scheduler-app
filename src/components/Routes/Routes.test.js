import React from 'react'
import { shallow } from 'enzyme'
import Routes from './Routes'

describe('<Routes>', () => {
    it('renders without errors', () => {
        shallow(<Routes />)
    })
})
