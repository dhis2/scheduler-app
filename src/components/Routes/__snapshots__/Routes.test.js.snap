// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes> renders correctly 1`] = `
<HashRouter>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(UnconnectedJobListContainer)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/edit/:id"
  />
  <Route
    component={[Function]}
    path="/add"
  />
</HashRouter>
`;
